### PR TITLE
Fix an edge case in OSSA RLE for loops

### PR DESCRIPTION
**Pull request by meg-gupta**
In OSSA RLE for loops, in certain cases SSAUpdater will not create a new
SILPhiArgument to be used as the forwarding value. Based on dominator info
it may return the newly copied available value as the forwarding value.
This newly copied available value in the dominating predecessor
will have destroy values at leaking blocks.
Instead of calling makeNewValueAvailable to make this newly copied value available
at the redundant load block, use makeExistingValueAvailable so that only
additional required destroy_values are inserted.

### CODE REVIEW

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci test macOS Platform

File: lib/SILOptimizer/Utils/InstOptUtils.cpp [hunk 1]
```diff
@@ -2025,12 +2025,31 @@ SILValue swift::makeNewValueAvailable(SILValue value, SILBasicBlock *inBlock) {
   assert(value->getUses().empty() &&
          value.getOwnershipKind() == OwnershipKind::Owned);
 
+  return makeExistingValueAvailable(value, inBlock);
+}
+
+SILValue swift::makeExistingValueAvailable(SILValue value,
+                                           SILBasicBlock *inBlock) {
```
Comment (eeckstein) [inline lib/SILOptimizer/Utils/InstOptUtils.cpp:2032]: This function is equivalent to `makeNewValueAvailable` (except the `assert(value->getUses().empty())`).
Why do we need a separate function? I looks like a simple rename to `makeExistingValueAvailable` of the old function would be sufficient.

File: lib/SILOptimizer/Transforms/RedundantLoadElimination.cpp [hunk 2]
```diff
@@ -1351,6 +1351,9 @@ SILValue RLEContext::computePredecessorLocationValue(SILBasicBlock *BB,
   }
 
   auto Val = Updater.getValueInMiddleOfBlock(BB);
+  if (!Val->use_empty()) {
```
Comment (eeckstein) [inline lib/SILOptimizer/Transforms/RedundantLoadElimination.cpp:1354]: Is this needed given that `makeExistingValueAvailable` does the same as `makeNewValueAvailable` (see my other comment)?

**Review by eeckstein — CHANGES_REQUESTED** [review]

File: lib/SILOptimizer/Utils/InstOptUtils.cpp (hunk 1 above)
Comment (meg-gupta) [reply lib/SILOptimizer/Utils/InstOptUtils.cpp:2032]: Yes, they are essentially the same except `makeExistingValueAvailable `  needs to consider uses. I added `makeExistingValueAvailable ` as a separate function to make this distinction. But it's probably making it more complex. I'll consolidate them.

**Comment by meg-gupta** [comment]
@swift-ci test

**Review by eeckstein — APPROVED** [review]
LGTM!

**Pull request merged**